src/shared/extensionUtilities.ts

```diff
@@ -123,3 +125,58 @@ function convertExtensionRootTokensToPath(
 ): string {
     return text.replace(/!!EXTENSIONROOT!!/g, `vscode-resource:${basePath}`)
 }
+
+/**
+ * Utility function to determine if the extension version has changed between activations
+ * False (versions are identical) if version key exists in global state, is a semver, and matches the current version
+ * True (versions are different) if any of the above aren't true
+ *
+ * TODO: Change the threshold on which we display the welcome page?
+ * For instance, if we start building nightlies, only show page for significant updates?
+ *
+ * @param context VS Code Extension Context
+ * @param currVersion Current version to compare stored most recent version against (useful for tests)
+ */
+export function isDifferentVersion(context: vscode.ExtensionContext, currVersion: string = pluginVersion): boolean {
+    const mostRecentVersion = context.globalState.get<string>(mostRecentVersionKey)
+    if (mostRecentVersion && semver.valid(mostRecentVersion) && mostRecentVersion === currVersion) {
```

Comment:
What would the impact be if the version was not a valid semver? Seems like that check isn't necessary.

Reply:
It isn't. I was thinking I'd compare semver versions to see if the plugin version was newer, but I'm instead opting to check if they're different (in case someone downgrades for some reason).